cmd/recorder/config/config.go

```diff
@@ -136,9 +161,88 @@ func (cfg *RecorderConfig) SetDefaults() {
 	if cfg.OutputFormat == "" {
 		cfg.OutputFormat = OutputFormatDefault
 	}
+
+	if cfg.VideoPreset == "" {
+		cfg.VideoPreset = VideoPresetDefault
+	}
+}
+
+func (cfg RecorderConfig) ToEnv() []string {
+	return []string{
+		fmt.Sprintf("SITE_URL=%s", cfg.SiteURL),
+		fmt.Sprintf("CALL_ID=%s", cfg.CallID),
+		fmt.Sprintf("THREAD_ID=%s", cfg.ThreadID),
+		fmt.Sprintf("AUTH_TOKEN=%s", cfg.AuthToken),
+		fmt.Sprintf("WIDTH=%d", cfg.Width),
+		fmt.Sprintf("HEIGHT=%d", cfg.Height),
+		fmt.Sprintf("VIDEO_RATE=%d", cfg.VideoRate),
+		fmt.Sprintf("AUDIO_RATE=%d", cfg.AudioRate),
+		fmt.Sprintf("FRAME_RATE=%d", cfg.FrameRate),
+		fmt.Sprintf("VIDEO_PRESET=%s", cfg.VideoPreset),
+		fmt.Sprintf("OUTPUT_FORMAT=%s", cfg.OutputFormat),
+	}
+}
+
+func (cfg RecorderConfig) ToMap() map[string]any {
+	return map[string]any{
+		"site_url":      cfg.SiteURL,
+		"call_id":       cfg.CallID,
+		"thread_id":     cfg.ThreadID,
+		"auth_token":    cfg.AuthToken,
+		"width":         cfg.Width,
+		"height":        cfg.Height,
+		"video_rate":    cfg.VideoRate,
+		"audio_rate":    cfg.AudioRate,
+		"frame_rate":    cfg.FrameRate,
+		"video_preset":  cfg.VideoPreset,
+		"output_format": cfg.OutputFormat,
+	}
 }
 
-func loadConfig() (RecorderConfig, error) {
+func (cfg *RecorderConfig) FromMap(m map[string]any) *RecorderConfig {
+	cfg.SiteURL, _ = m["site_url"].(string)
+	cfg.CallID, _ = m["call_id"].(string)
+	cfg.ThreadID, _ = m["thread_id"].(string)
+	cfg.AuthToken, _ = m["auth_token"].(string)
+	if width, ok := m["width"].(float64); ok {
+		cfg.Width = int(width)
+	} else {
+		cfg.Width, _ = m["width"].(int)
+	}
```

Comment:
Or we fail for inputs with decimals? User input handling is fun.

Reply:
lol, that's pretty horrible code necessary since Go doesn't support unions plus the way structs get serialized from the plugin is messed. Numeric values get automatically converted as `float64` when converting to/from `interface{}`.